### PR TITLE
SLVSCODE-21 Enable TypeScript analysis

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -87,17 +93,17 @@
     "server/sonarlint-ls.jar"
   ],
   "scripts": {
-    "vscode:prepublish": "tsc -p ./ && node scripts/prepare.js",
-    "compile": "tsc -watch -p ./",
+    "vscode:prepublish": "./node_modules/.bin/tsc -p ./ && node scripts/prepare.js",
+    "compile": "./node_modules/.bin/tsc -watch -p ./",
```
Comment (vilchik-elena) [inline package.json]: @henryju do we need this `watch`?

File: scripts/prepare.js [hunk 2]
```diff
@@ -27,6 +27,10 @@ downloadIfNeeded(
   "http://repo1.maven.org/maven2/org/sonarsource/python/sonar-python-plugin/1.8.0.1496/sonar-python-plugin-1.8.0.1496.jar",
   "analyzers/sonarpython.jar"
 );
+downloadIfNeeded(
+  "https://repox.sonarsource.com/sonarsource/org/sonarsource/typescript/sonar-typescript-plugin/1.5.0.2021/sonar-typescript-plugin-1.5.0.2021.jar",
```
Comment (vilchik-elena) [inline scripts/prepare.js]: @henryju we will update these build numbers (ts and sonarlint-ls) before release, right?

File: src/extension.ts [hunk 3]
```diff
@@ -126,18 +131,61 @@ function toUrl(filePath) {
...
+
+  return Path.join(VSCode.workspace.rootPath, tsdkPathSetting);
+}
+
 export function activate(context: VSCode.ExtensionContext) {
   let serverOptions = () => runJavaServer(context);
 
+  const tsExt = VSCode.extensions.getExtension("vscode.typescript");
+  if (!tsExt) {
+    console.warn(
```
Comment (vilchik-elena) [inline src/extension.ts]: @henryju should we return after this `warn`?

File: src/extension.ts [hunk 4]
```diff
@@ -126,18 +131,61 @@ function toUrl(filePath) {
...
+      "Unable to locate TypeScript extension. No TypeScript support in SonarLint"
+    );
+  }
+  const tsdkPathSetting = VSCode.workspace
+    .getConfiguration("typescript")
+    .get("tsdk", undefined);
```
Comment (vilchik-elena) [inline src/extension.ts]: @henryju omit second parameter, it will do the same job

File: package.json (hunk 1 above)
Comment (henryju) [reply package.json]: No idea. It certainly come from the original template we used as a quick starter. Do you suggest we should remove it?

Comment (vilchik-elena) [reply package.json]: @henryju in fact it does nothing:) it should be `-w` or `--watch`. So you can remove it.

**Pull request merged**